packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -38,25 +48,106 @@ function isValidSpanId(spanId: string): boolean {
   return VALID_SPANID_REGEX.test(spanId) && !INVALID_ID_REGEX.test(spanId);
 }
 
+function isValidParentSpanID(spanId: string | undefined): boolean {
+  return spanId === undefined || isValidSpanId(spanId);
+}
+
+function isValidSampledValue(sampled: number | undefined): boolean {
+  return (
+    sampled === undefined ||
+    VALID_SAMPLED_VALUES.includes(sampled) ||
+    VALID_UNSAMPLED_VALUES.includes(sampled)
+  );
+}
+
+function parseHeader(header: unknown) {
+  return Array.isArray(header) ? header[0] : header;
+}
+
 /**
  * Propagator for the B3 HTTP header format.
  * Based on: https://github.com/openzipkin/b3-propagation
  */
 export class B3Propagator implements HttpTextPropagator {
+  _getHeaderValue(carrier: unknown, getter: GetterFunction, key: string) {
+    const header = getter(carrier, key);
+    return parseHeader(header);
+  }
+
+  _getTraceId(carrier: unknown, getter: GetterFunction): string {
+    const traceId = this._getHeaderValue(carrier, getter, X_B3_TRACE_ID);
+    if (typeof traceId === 'string') {
+      return traceId.padStart(32, '0');
+    }
+    return '';
+  }
+
+  _getSpanId(carrier: unknown, getter: GetterFunction): string {
+    const spanId = this._getHeaderValue(carrier, getter, X_B3_SPAN_ID);
+    if (typeof spanId === 'string') {
+      return spanId;
+    }
+    return '';
+  }
+
+  _getParentSpanId(
+    carrier: unknown,
+    getter: GetterFunction
+  ): string | undefined {
+    const spanId = this._getHeaderValue(carrier, getter, X_B3_PARENT_SPAN_ID);
+    if (typeof spanId === 'string') {
+      return spanId;
+    }
+    return;
+  }
+
+  _getDebug(carrier: unknown, getter: GetterFunction): string | undefined {
+    const debug = this._getHeaderValue(carrier, getter, X_B3_FLAGS);
+    return debug === '1' ? '1' : undefined;
+  }
+
+  _getTraceFlags(carrier: unknown, getter: GetterFunction): number | undefined {
+    const traceFlags = this._getHeaderValue(carrier, getter, X_B3_SAMPLED);
+    const debug = this._getDebug(carrier, getter);
+    if (debug === '1') {
+      return TraceFlags.SAMPLED;
+    } else if (traceFlags !== undefined) {
+      if (VALID_SAMPLED_VALUES.includes(traceFlags)) {
+        return TraceFlags.SAMPLED;
+      } else if (VALID_UNSAMPLED_VALUES.includes(traceFlags)) {
+        return TraceFlags.NONE;
+      } else {
+        // Invalid traceflag
+        return 2;
```

Comment:
The second bit has no defined meaning. What is the reason for setting this undefined flag?

Reply:
I was just using this to mark it as an invalid traceflag so it could never hit the `setExtractedSpanContext` branch